src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -1639,6 +1639,7 @@ Context* AbstractWriteLog<I>::construct_flush_entry(std::shared_ptr<GenericLogEn
     m_lowest_flushing_sync_gen = log_entry->ram_entry.sync_gen_number;
   }
   m_flush_ops_in_flight += 1;
+  m_flush_ops_will_send += 1;
```

Comment:
@majianpeng  can you mention in the commit:
1. why we need this reordering in this commit as well, just so that if someone visits it, we still be able to get the context.
2. Is this solving some additional problem apart from fix in https://github.com/ceph/ceph/pull/43127 or just improving on it?

Reply:
@ideepika . In the process of fix issue https://tracker.ceph.com/issues/53108,  i found this problem.  It only fix pwl cache reorder problem. we make sure issue ops by order in pwl cache.   https://github.com/ceph/ceph/pull/43127 to make sure ops order in the func construct_flush_entries. But it don't consider order between construct_flush_entries.
This pr is to fix this problem

Reply:
@majianpeng are you working on https://tracker.ceph.com/issues/53108 ? 
http://qa-proxy.ceph.com/teuthology/ideepika-2021-11-12_08:56:59-rbd-wip-deepika-testing-2021-11-12-1203-distro-basic-smithi/6499122/teuthology.log

Reply:
Yes, i'm currently solving. I found two races in objectmap which cause reorder but it still can't fix issue53108. I doubt there is other problem.

Reply:
I see, not much well versed in this part myself.
Good to know it's in progress, thanks for the update!